crm/server-scripts/crm/Opportunity/UpdateCampaigns.js

```diff
@@ -0,0 +1,47 @@
+export default {
+  label: 'Update Campaigns after creation or update',
+  description: 'Update related Campaigns after a Opportunity is created or updated',
+
+  * triggers ({ after }) {
+    yield after('create', 'update')
+      .for('compose:record')
+      .where('module', 'Opportunity')
+      .where('namespace', 'crm')
+  },
+
+  async exec ({ $record }, { Compose }) {
+    // Update campaign record
+    if ($record.values.CampaignId) {
+      await Compose.findRecords(`recordID = '${$record.values.CampaignId}'`, 'Campaigns')
+        .catch(() => ({ set: [] }))
+        .then(({ set }) => {
+          Promise.all(set.map(campaign => {
+            // Find all Opportunities that have the same Campaign
+            const { set, filter } = await Compose.findRecords(`CampaignId = '${campaign.recordID}'`, 'Opportunity')
+              .catch(() => ({ set: [], filter: { count: 0 } }))
+
+            let numWon = 0
+            let amount = 0
+            let amountWon = 0
+
+            set.forEach(o => {
+              amount += o.values.Amount || 0
+              if (o.values.IsWon) {
+                numWon += 1
+                amountWon += o.values.Amount || 0
+              }
+            })
+
+            campaign.values.NumberOfOpportunities = filter.count
+            campaign.values.NumberOfWonOpportunities = numWon
+            campaign.values.AmountAllOpportunities = amount
+            campaign.values.AmountWonOpportunities = amountWon
+
+            return Compose.saveRecord(campaign)
```

Comment:
Same here, make multiple queries to the API.

The main reasons:

* There can be a lot of data; its a waste of resources to needlessly handle them
* There can be a lot of data; you would need to do it in chunks so the system doesn't just die